--- a/net/minecraft/src/Block.java
+++ b/net/minecraft/src/Block.java
@@ -58,7 +58,9 @@
 	public static final Block clothWhite = null;
 	public static final MouseHelper ae = (MouseHelper)(new MouseHelper(37, 13)).setHardness(0.0F).setStepSound(soundGrassFootstep);
 	public static final MouseHelper af = (MouseHelper)(new MouseHelper(38, 12)).setHardness(0.0F).setStepSound(soundGrassFootstep);
+	public static final MouseHelper grassFlower = (MouseHelper)(new MouseHelper(125, 14)).setHardness(0.0F).setStepSound(soundGrassFootstep);
 	public static final MouseHelper ag = (MouseHelper)(new BlockMushroom(39, 29)).setHardness(0.0F).setStepSound(soundGrassFootstep).setLightValue(0.125F);
+	public static final MouseHelper glowShroom = (MouseHelper)(new BlockMushroom(127, 30)).setHardness(0.0F).setStepSound(soundGrassFootstep).setLightValue(0.9375F);
 	public static final MouseHelper ah = (MouseHelper)(new BlockMushroom(40, 28)).setHardness(0.0F).setStepSound(soundGrassFootstep);
 	public static final Block blockGold = (new BlockOreBlock(41, 39)).setHardness(3.0F).setResistance(10.0F).setStepSound(soundMetalFootstep);
 	public static final Block blockSteel = (new BlockOreBlock(42, 38)).setHardness(5.0F).setResistance(10.0F).setStepSound(soundMetalFootstep);
@@ -134,6 +136,7 @@
 	public static final Block imgur9F0A3un = (new Block(102, 157, Material.rock)).setHardness(2.0F).setResistance(10.0F).setStepSound(soundStoneFootstep);
 	public static final Block yeahWeHadFunWithThisAsYouCanSee = (new Block(103, 158, Material.rock)).setHardness(2.0F).setResistance(10.0F).setStepSound(soundStoneFootstep);
 	public static final Block BarrierBlock = (new BlockHidable(104, 159, Material.glass, false)).setHardness(0.3F).setStepSound(soundGlassFootstep);
+	public static final Block PurpleDev = (new Block(122, 160, Material.rock)).setHardness(2.0F).setResistance(10.0F).setStepSound(soundStoneFootstep);
 	public static final Block StairLadderBlock = (new BlockLadder(105, 161)).setHardness(0.4F).setStepSound(soundWoodFootstep);
 	public static final Block fakeDirt = (new Block(106, 162, Material.rock)).setHardness(0.6F).setResistance(10.0F).setStepSound(soundGrassFootstep);
 	public static final Block fakeRock = (new Block(107, 163, Material.rock)).setHardness(2.0F).setResistance(10.0F).setStepSound(soundStoneFootstep);
@@ -145,11 +148,17 @@
 	public static final Block dbg2Block = (new BlockGlowing(113, 150, 1)).setHardness(0.0F).setLightValue(0.9375F).setStepSound(soundStoneFootstep);
 	public static final Block saltBlock = (new BlockMultiSided(114, 170, 169, 170)).setHardness(0.6F).setStepSound(soundStoneFootstep);
 	public static final Block glowingFlower = (new BlockGlowingFlower(115, 171, 1)).setHardness(0.0F).setLightValue(0.9375F).setStepSound(soundGrassFootstep);
+	public static final Block glowingFlowerWhite = (new BlockGlowingFlower(124, 183, 1)).setHardness(0.0F).setLightValue(0.9375F).setStepSound(soundGrassFootstep);
 	public static final Block glowingFlowerInfBasic = (new BlockGlowingFlowerInfused(117, 173, 1, 2)).setHardness(0.0F).setLightValue(0.9375F).setStepSound(soundGrassFootstep);
 	public static final Block glowingFlowerInfGold = (new BlockGlowingFlowerInfused(118, 174, 1, 4)).setHardness(0.0F).setLightValue(0.9375F).setStepSound(soundGrassFootstep);
 	public static final Block glowingFlowerInfObsidian = (new BlockGlowingFlowerInfused(119, 175, 1, 7)).setHardness(0.0F).setLightValue(0.9375F).setStepSound(soundGrassFootstep);
 	public static final Block blueFireIdk = (new BlockGlowing(116, 172, 2)).setHardness(0.0F).setLightValue(0.9375F).setStepSound(soundStoneFootstep);
 	public static final Block safe = (new BlockSafe(120, 177, 176, 177)).setHardness(1.5F).setResistance(10.0F).setStepSound(soundStoneFootstep);
+	//public static final Block ashInfuser = (new Block(121, 179, Material.rock)).setHardness(0.6F).setStepSound(soundStoneFootstep);
+	//public static final Block cuttingLathe = (new BlockLathe(123, 181, 180, 182)).setHardness(2.0F).setResistance(10.0F).setStepSound(soundStoneFootstep);
+	public static final Block gemLog = (new BlockGemLog(121)).setHardness(2.0F).setStepSound(soundWoodFootstep);
+	public static final Block gemPlanks = (new Block(123, 78, Material.wood)).setHardness(2.0F).setResistance(5.0F).setStepSound(soundWoodFootstep);
+	public static final Block gemLeaves = (new BlockGlass(126, 79, Material.grass, false)).setHardness(0.2F).setLightOpacity(1).setStepSound(soundGrassFootstep);
 
 	protected Block(int id, Material material) {
 		this.stepSound = soundPowderFootstep;
