--- a/net/minecraft/src/BlockTNT.java
+++ b/net/minecraft/src/BlockTNT.java
@@ -30,8 +30,8 @@
 	}
 
 	public void onBlockDestroyedByPlayer(World worldObj, int x, int y, int z, int metadata) {
-		EntityTNTPrimed entityTNTPrimed6 = new EntityTNTPrimed(worldObj, (float)x + 0.5F, (float)y + 0.5F, (float)z + 0.5F);
-		worldObj.spawnEntityInWorld(entityTNTPrimed6);
-		worldObj.playSoundAtEntity(entityTNTPrimed6, "random.fuse", 1.0F, 1.0F);
+		//EntityTNTPrimed entityTNTPrimed6 = new EntityTNTPrimed(worldObj, (float)x + 0.5F, (float)y + 0.5F, (float)z + 0.5F);
+		//worldObj.spawnEntityInWorld(entityTNTPrimed6);
+		//worldObj.playSoundAtEntity(entityTNTPrimed6, "random.fuse", 1.0F, 1.0F);
 	}
 }
