--- a/net/minecraft/src/BlockFire.java
+++ b/net/minecraft/src/BlockFire.java
@@ -2,6 +2,7 @@
 
 import java.io.DataInputStream;
 import java.io.DataOutputStream;
+import java.io.IOException;
 
 public class BlockFire extends Packet {
 	public String a;
@@ -13,11 +14,11 @@
 		this.a = string1;
 	}
 
-	public void readPacketData(DataInputStream dataInputStream) {
+	public void readPacketData(DataInputStream dataInputStream) throws IOException {
 		this.a = dataInputStream.readUTF();
 	}
 
-	public void writePacket(DataOutputStream dataOutputStream) {
+	public void writePacket(DataOutputStream dataOutputStream) throws IOException {
 		dataOutputStream.writeUTF(this.a);
 	}
 
