--- a/net/minecraft/src/GuiMultiplayer.java
+++ b/net/minecraft/src/GuiMultiplayer.java
@@ -65,7 +65,7 @@
 	}
 
 	public void drawScreen(int mouseX, int mouseY, float renderPartialTick) {
-		this.drawDefaultBackground();
+		this.drawOptionBackground();
 		this.drawCenteredString(this.fontRenderer, "Play Multiplayer", this.width / 2, this.height / 4 - 60 + 20, 0xFFFFFF);
 		this.drawString(this.fontRenderer, "Minecraft Multiplayer is currently not finished, but there", this.width / 2 - 140, this.height / 4 - 60 + 60 + 0, 10526880);
 		this.drawString(this.fontRenderer, "is some buggy early testing going on.", this.width / 2 - 140, this.height / 4 - 60 + 60 + 9, 10526880);
