--- a/net/minecraft/src/ItemFlintAndSteel.java
+++ b/net/minecraft/src/ItemFlintAndSteel.java
@@ -31,7 +31,7 @@
 
 	}
 
-	public void readPacketData(DataInputStream dataInputStream) {
+	public void readPacketData(DataInputStream dataInputStream) throws IOException {
 		this.a = dataInputStream.readInt();
 		this.b = dataInputStream.readShort();
 		this.c = dataInputStream.readInt();
@@ -41,7 +41,7 @@
 		this.e = CompressedStreamTools.decompress(this.d);
 	}
 
-	public void writePacket(DataOutputStream dataOutputStream) {
+	public void writePacket(DataOutputStream dataOutputStream) throws IOException {
 		dataOutputStream.writeInt(this.a);
 		dataOutputStream.writeShort(this.b);
 		dataOutputStream.writeInt(this.c);
