--- a/net/minecraft/src/PlayerControllerCreative.java
+++ b/net/minecraft/src/PlayerControllerCreative.java
@@ -9,11 +9,11 @@
 	}
 
 	public void onRespawn(EntityPlayer entityPlayer) {
-		for(int i2 = 0; i2 < 9; ++i2) {
+		for(int i2 = 0; i2 < 64; ++i2) {
 			if(entityPlayer.inventory.mainInventory[i2] == null) {
 				this.mc.thePlayer.inventory.mainInventory[i2] = new ItemStack(((Block)Session.registeredBlocksList.get(i2)).blockID);
 			} else {
-				this.mc.thePlayer.inventory.mainInventory[i2].stackSize = 1;
+				this.mc.thePlayer.inventory.mainInventory[i2].stackSize = 64;
 			}
 		}
 
