--- a/net/minecraft/src/GuiIngame.java
+++ b/net/minecraft/src/GuiIngame.java
@@ -281,6 +281,9 @@
 			this.drawString(fontRenderer8, string36, i6 - fontRenderer8.getStringWidth(string36) - 2, 12, 14737632);
 		} else {
 			fontRenderer8.drawStringWithShadow("Minecraft Alpha 1.0.16.05_13 Lilypad", 2, 2, 0xFFFFFF);
+			this.drawString(fontRenderer8, "x: " + this.mc.thePlayer.posX, 2, 14, 14737632);
+			this.drawString(fontRenderer8, "y: " + this.mc.thePlayer.posY, 2, 26, 14737632);
+			this.drawString(fontRenderer8, "z: " + this.mc.thePlayer.posZ, 2, 38, 14737632);
 			if(uqKey != "") {
 				fontRenderer8.drawStringWithShadow(uqKey, 2, 12, 0xFFFFFF);
 			}
@@ -330,19 +333,6 @@
 		GL11.glDisable(GL11.GL_ALPHA_TEST);
 		GL11.glPushMatrix();
 		GL11.glTranslatef(0.0F, (float)(i7 - 48), 0.0F);
-		if(GuiScreen.currentID != "") {
-			fontRenderer8.drawStringWithShadow("[Use numpad to enter, - to clear, + to give]", 2, 10, 16449390);
-			fontRenderer8.drawStringWithShadow("Give item: " + GuiScreen.currentID, 2, 20, 16449390);
-
-			try {
-				i30 = Integer.parseInt(GuiScreen.currentID);
-				if((Block.blocksList.length <= i30 || Block.blocksList[i30] == null) && (Item.itemsList.length <= i30 || Item.itemsList[i30] == null)) {
-					fontRenderer8.drawStringWithShadow("(INVALID)", 2, 30, 16711680);
-				}
-			} catch (Exception exception24) {
-				fontRenderer8.drawStringWithShadow("(ERROR)", 2, 30, 16711680);
-			}
-		}
 
 		for(i30 = 0; i30 < this.chatMessageList.size() && i30 < b29; ++i30) {
 			if(((ChatLine)this.chatMessageList.get(i30)).updateCounter < 200 || z34) {
