--- /dev/null
+++ b/net/minecraft/src/GuiSetting.java
@@ -1,0 +1,17 @@
+package net.minecraft.src;
+
+public class GuiSetting extends GuiScreen {
+
+	public void initGui() {
+		this.controlList.add(new GuiButton(1, this.width / 2 - 100, this.height / 4 + 72, "Singleplayer"));
+	}
+
+	protected void actionPerformed(GuiButton button) {
+
+	}
+
+	public void drawScreen(int mouseX, int mouseY, float renderPartialTick) {
+		this.drawDefaultBackground();
+		super.drawScreen(mouseX, mouseY, renderPartialTick);
+	}
+}
