--- a/net/minecraft/src/InputHandler.java
+++ b/net/minecraft/src/InputHandler.java
@@ -32,7 +32,7 @@
 	private static float lastRotX;
 	private static float lastRotZ;
 	public static int lookSens = 5;
-	public static boolean cheatsEnabled = true;
+	public static boolean cheatsEnabled = net.minecraft.src.GameSettings.cheats;
 	public static int KBEventKey = -1;
 	public static boolean KBEventKeyState = false;
 	public static boolean LastAPress = false;
