--- a/net/minecraft/src/RecipesWeapons.java
+++ b/net/minecraft/src/RecipesWeapons.java
@@ -2,6 +2,7 @@
 
 import java.io.DataInput;
 import java.io.DataOutput;
+import java.io.IOException;
 
 public class RecipesWeapons extends NBTBase {
 	public String a;
@@ -16,11 +17,11 @@
 		}
 	}
 
-	void writeTagContents(DataOutput dataOutput) {
+	void writeTagContents(DataOutput dataOutput) throws IOException {
 		dataOutput.writeUTF(this.a);
 	}
 
-	void readTagContents(DataInput dataInput) {
+	void readTagContents(DataInput dataInput) throws IOException {
 		this.a = dataInput.readUTF();
 	}
 
