--- a/net/minecraft/src/ScreenInputPass.java
+++ b/net/minecraft/src/ScreenInputPass.java
@@ -130,8 +130,8 @@
 			this.drawString(this.fontRenderer, this.i + (this.h / 6 % 2 == 0 ? "_" : ""), i4 + 4, i5 + 6, 14737632);
 			this.drawString(this.fontRenderer, this.einval, this.width / 2 - 120, this.height / 4 - 60 + 60 + 90, 16711680);
 		} else {
-			this.drawCenteredString(this.fontRenderer, "Welcome back to Lilypad", this.width / 2, this.height / 4 - 60 + 20, -1593835521);
-			this.drawString(this.fontRenderer, "Remember to report bugs to the bug tracker", this.width / 2 - 140, this.height / 4 - 60 + 60 + 0, 10526880);
+			this.drawCenteredString(this.fontRenderer, "Welcome back to the Afterglow Client", this.width / 2, this.height / 4 - 60 + 20, -1593835521);
+			this.drawString(this.fontRenderer, "Remember to report bugs to the Afterglow github page", this.width / 2 - 140, this.height / 4 - 60 + 60 + 0, 10526880);
 		}
 
 		super.drawScreen(mouseX, mouseY, renderPartialTick);
